### ISSUE TITLE
PDF reader: Title of Contents extraction

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#2285]
ZotFile has this. I have no idea how widely it's used, or what it's used for. Maybe people extract a ToC and then use that as a template for notes?

Is this the same (in content, not in usage) as the Document Outline tab in the annotation sidebar?

**Comment by mrtcode** [comment]
Yes, it's the same "Document Outline", and yes ToC is probably used as a template that is filled with the most important content later. I guess it helps people to think. Especially for books, as ToCs are more common between them.

If we want this, it shouldn't be difficult to add somewhere a context menu option to create a note from a ToC or copy to clipboard. Maybe a context menu option on ToC tab in pdf-reader.

In theory it would be possible to re-create ToC from PDF text even if it doesn't exist...

**Comment by dstillman** [comment]
> Maybe a context menu option on ToC tab in pdf-reader.

I don't think anyone would find that. Just a "Create Note from Outline" button at the bottom of that tab?

**Comment by dstillman** [comment]
https://forums.zotero.org/discussion/89682/how-to-extract-annotations-along-with-the-table-of-contents

This actually seems more useful: an option to include ToC headers when adding annotations to notes. Still not sure how many people would want it, but it might be neat as an option if we can do it fairly easily.

**Comment by mrtcode** [comment]
Actually it isn't that easy, because I can only get a point to the PDF page where the header approximately is. And I need to determine the text offset of the header.

Therefore I would be more inclined to implement outline extraction directly from text and only use the actual ToC to validate the result.
